internal/controller/taggable_controller.go

```diff
@@ -4,24 +4,28 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
 
 	ccv1alpha1 "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
 	tagsv1alpha1 "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/tags/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/deliveryhero/gcp-config-connector-tagging-operator/internal/gcp"
 )
 
 const (
-	projectIDAnnotation = "cnrm.cloud.google.com/project-id"
-	tagBindingOwnerKey  = ".metadata.controller"
+	projectIDAnnotation       = "cnrm.cloud.google.com/project-id"
+	tagBindingOwnerKey        = ".metadata.controller"
+	taggableResourceFinalizer = "tagging.gcp.dhero.io/finalizer" // Add finalizer constant
```

Comment:
```suggestion
	taggableResourceFinalizer = "gdp.deliveryhero.io/resource-tags"
```

To make it more consistent with the field-exporter project